packages/flutter/lib/src/rendering/proxy_box.dart

```diff
@@ -1471,11 +1418,199 @@ class RenderPhysicalModel extends _RenderCustomClip<RRect> {
     }
   }
 
+  @override
+  bool hitTest(HitTestResult result, { Offset position }) {
+    if (_clipper != null) {
+      _updateClip();
+      assert(_clip != null);
+      if (!_clip.contains(position))
+        return false;
+    }
+    return super.hitTest(result, position: position);
+  }
+
   @override
   void debugFillProperties(DiagnosticPropertiesBuilder description) {
     super.debugFillProperties(description);
     description.add(new DiagnosticsProperty<BoxShape>('shape', shape));
     description.add(new DiagnosticsProperty<BorderRadius>('borderRadius', borderRadius));
+  }
+}
+
+/// Creates a physical model layer that clips its children to a [ShapeBorder].
+///
+/// A physical model layer casts a shadow based on its [elevation].
```

Comment:
I think you want to refer to this (here and elsewhere) as a "physical shape".

Reply:
Done.
I kept the model term here to stay somewhat consistent with RenderPhysicalModel, but I totally get your point.